ring/func/base.py

```diff
@@ -104,27 +104,31 @@ def coerce_function(t):
     # but NEVER add a general iterator processing. it will cause user bugs.
 
 
-def coerce(v):
+def coerce(v, in_memory_storage):
     """Transform the given value to cache-friendly string data."""
+
     type_coerce = coerce_function(type(v))
     if type_coerce:
         return type_coerce(v)
 
     if hasattr(v, '__ring_key__'):
         return v.__ring_key__()
 
+    if in_memory_storage and type(v).__hash__ != object.__hash__:
+        return hash(v)
```

Comment:
Becasue `hash()` only returns simple integer, It must be prefixed with specific type name. look into `ring._compat.qualname` to generate a qualname for python2/3.

One of preferred generated key is:
`A(hash:49320493290432)` or `A:hash:49320493290432`